bin/spice/cmd/refresh.go

```diff
@@ -17,17 +17,46 @@ limitations under the License.
 package cmd
 
 import (
+	"encoding/json"
 	"fmt"
 
 	"github.com/spf13/cobra"
 	"github.com/spiceai/spiceai/bin/spice/pkg/api"
 	"github.com/spiceai/spiceai/bin/spice/pkg/context"
+	"github.com/spiceai/spiceai/bin/spice/pkg/spec"
+)
+
+const (
+	refreshSqlFlag  = "refresh-sql"
+	refreshModeFlag = "refresh-mode"
 )
 
 type DatasetRefreshApiResponse struct {
 	Message string `json:"message,omitempty"`
 }
 
+type DatasetRefreshApiRequest struct {
+	SQL  *string `json:"refresh_sql,omitempty"`
+	Mode *string `json:"refresh_mode,omitempty"`
+}
+
+func construct_request(sql string, mode string) (*string, error) {
```

Comment:
```suggestion
func constructRequest(sql string, mode string) (*string, error) {
```

Let's stick to the Go style function formatting in Go